EVM-verifier/contracts/AliceRingToken.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract RingSignatureVerifier {
+    constructor() {}
+
+    function verify(string memory signature) public pure returns (bool) {
+        return true;
+    }
+}
+
+contract AliceRingToken is ERC721, ERC721URIStorage, Ownable {
+    uint256 private _nextTokenId;
+
+    struct pubKey {
+        uint256 x;
+        uint256 y;
+    }
+
+    struct Signature {
+        pubKey[] ring;
+        uint256 c;
+        string message;
+        uint256[] responses;
+    }
+
+    enum Status {
+        UNKNOWN, // the proof has not been verified on-chain, no proof has been minted
+        MINTED // the proof has already been minted
+    }
+
+    error AlreadyMinted(string proofId);
+    error InvalidSignature();
+    error InvalidTokenAmounts();
+    error OnlyOwnerCanBurn(uint256 tokenId);
+
+    RingSignatureVerifier public verifier; // Instance of the ring signature verifier contract
+    uint256 public minimalAmount; // Minimal amount of tokens to be owned by each address in the ring
+
+    mapping(string => Status) public mintStatus; // signatureHash => Status (computed off-chain)
+
+    constructor(address initialOwner, address _verifier, uint256 _minimalAmount)
+        ERC721("AliceRingToken", "ART")
+        Ownable(initialOwner)

```

Comment:
`Ownable`  does not take any argument

Reply:
It does in the latest v5 implementation of Openzeppelin. If you use an earlier version, then there's no argument.
Check this out - https://blog.openzeppelin.com/introducing-openzeppelin-contracts-5.0